app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -299,96 +299,104 @@ def vms
         parent_cluster = nil
       end
 
-      availability_zone = s.availability_zone.blank? ? "null_az" : s.availability_zone
```

Comment:
Hm cutting this method here instead of above to work around the shared `hosts` looks awkward.

Prefer:
```ruby
collector.vms.each do |server|
  parse_server(server)
end
```

And deal with the hosts another way like move it to the collector or add a parser instance variable